src/include/migraphx/op/resize.hpp

```diff
@@ -0,0 +1,328 @@
+                                   [](auto size_i) { return size_i; });
+                    // Deduce the scales for each axis
+                    std::transform(
+                        input.begin(),
+                        input.end(),
+                        in_lens.begin(),
+                        vec_scale.begin(),
+                        [](auto sz, size_t in_len) { return static_cast<double>(sz) / in_len; });
+                    vec_scale[0] = 1.0f;
+                }
+                else
+                {
+                    // read the scale from args[1]
+                    //
+                    std::transform(input.begin(), input.end(), vec_scale.begin(), [](auto scale_i) {
+                        return scale_i;
+                    });
+                    // compute the output dimensions from the given scales.  This computation
+                    // always rounds down, unlike the internal computation in Nearest mode
+                    // which has several options as given in nearest_mode.
+                    std::transform(input.begin(),
+                                   input.end(),
+                                   in_lens.begin(),
+                                   out_lens.begin(),
+                                   [](auto scale_i, size_t in_len) {
+                                       return static_cast<size_t>(scale_i * in_len);
+                                   });
+                }
+            });
+        }
+        if(dyn_out.computed_shape.dynamic())
+        {
+            // at this time, case of single, dynamic input is not allowed.
+            output_shape = {args[0].get_shape().type(), out_lens};
+        }
+        else
+        {
+            output_shape = dyn_out.computed_shape;
+            if(not sizes.empty())
+            {
+                out_lens = sizes;
+                // compute scales
+                std::transform(out_lens.begin(),
+                               out_lens.end(),
+                               in_lens.begin(),
+                               vec_scale.begin(),
+                               [](size_t out_len, size_t in_len) {
+                                   return (in_len == 0 ? 1.f
+                                                       : static_cast<float>(out_len) / in_len);
+                               });
+            }
+            else
+            {
+                vec_scale = this->scales;
+                // compute output sizes
+                const auto& inputs = output_shape.lens();
+                std::transform(inputs.begin(),
+                               inputs.end(),
+                               scales.begin(),
+                               out_lens.begin(),
+                               [](auto scale_i, size_t in_len) {
+                                   return static_cast<size_t>(scale_i * in_len);
+                               });
+            }
+        }
```

Comment:
Given that this is a static input tensor we should be able to calculate the scales given the output sizes or vice versa at parsing. So we could just have both scales and sizes set during parsing and not have to do this calculation at runtime?

Reply:
If I remember, my original reason for doing it this way was to populate the local variable vec_scale from either inputs or attributes, while keeping the code that does it together.  Whether or not it's really more clear, I don't think the difference is worth refactoring at this point.  Computationally it's insignificant since the number of dimensions is small.

Also, recall that in the current configuration this code branch will never be exercised--the parser converts a Resize instruction with static inputs to the Gather op, not Resize.